sql-plugin/src/main/scala/com/nvidia/spark/rapids/RapidsConf.scala

```diff
@@ -1647,6 +1687,14 @@ class RapidsConf(conf: Map[String, String]) extends Logging {
   }
 
   private def getOptionalCost(key: String) = {
-    conf.get(key).map(toDouble(_, key))
+    // this is a bit convoluted, but the flow is to see if we have a
+    // defined configuration setting (potentially with a default value)
+    // and if not, then we look for the key in the configuration values
+    // anyway so that we can provide costs for any operator or expression,
+    // even if there is no well-defined configuration yet
+    OPTIMIZER_COSTS.get(key)
```

Comment:
If we're going to have to always special-case these settings to handle the defaults properly then I'd rather go back to the prior design.  The problem with this new approach is the need to maintain the conf in two places, the RapidsConf entry and the OPTIMIZER_COSTS list.  If someone forgets to update the latter then it will silently not honor the custom default which won't be fun to track down.